pkg/lib/authenticationflow/declarative/intent_login_flow_step_authenticate.go

```diff
@@ -40,7 +40,7 @@ func (i *IntentLoginFlowStepAuthenticate) GetJSONPointer() jsonpointer.T {
 
 var _ IntentLoginFlowStepChangePasswordTarget = &IntentLoginFlowStepAuthenticate{}
 
-func (i *IntentLoginFlowStepAuthenticate) GetPasswordAuthenticator(_ context.Context, _ *authflow.Dependencies, flows authflow.Flows) (info *authenticator.Info, ok bool) {
+func (i *IntentLoginFlowStepAuthenticate) GetPasswordAuthenticator(_ context.Context, _ *authflow.Dependencies, flows authflow.Flows) (info *authenticator.Info, changeRequired bool) {
```

Comment:
I don't think `ok` had any meaning other than password authenticator need update?

Reply:
🤔 Then it seems `GetPasswordAuthenticator` should be named as `GetChangeRequiredPasswordAuthenticator`, because it will not return the password authenticator if it doesn't require change, even it exist in the flow.

Reply:
Or maybe I move the info assign outside of the condition then

Reply:
Let's rename GetPasswordAuthenticator to GetChangeRequiredPasswordAuthenticator to match its behavior.